File: 0023-merge-k-sorted-lists/0023-merge-k-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeKLists(vector<ListNode*>& lists) {
        priority_queue<int,vector<int>,greater<int>>pq;
        int k=lists.size();
        for(int i=0;i<k;i++){
            ListNode* head=lists[i];
            while(head){
                pq.push(head->val);
                head=head->next;
            }
        }
         if (pq.empty()) {
            return nullptr; // If there are no nodes, return nullptr
        }
        ListNode* ans=new ListNode(pq.top());
        ListNode* temp=ans;
        pq.pop();
        while(!pq.empty()){
            ListNode* new_node=new ListNode(pq.top());
            temp->next=new_node;
            temp=temp->next;
            pq.pop();
        }
        return ans;
    }
};

File: 0023-merge-k-sorted-lists/0023-merge-k-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        List<Integer> list=new ArrayList<>();
        for(ListNode a:lists){
           f(list,a);
        }
        Collections.sort(list);
         if (list.isEmpty()) {
            return null; 
        }
        ListNode head = new ListNode(list.get(0));
        ListNode current = head;
        for (int i = 1; i < list.size(); i++) {
            current.next = new ListNode(list.get(i));
            current = current.next;
        }

        return head; 
    }
    public List<Integer> f(List<Integer> l,ListNode temp){
        while(temp!=null){
            l.add(temp.val);
            temp=temp.next;
        }
        return l;
    }
}